docs/types.rst

```diff
@@ -575,17 +589,40 @@ Default data location:
  - parameters (also return) of functions: memory
  - all other local variables: storage
 
+
+.. index:: copy an array, copy a struct
+
+Copying Reference Types
```

Comment:
I think this should also go into the newly created "copying, references, function calls, whatever" section with a link from here.

Reply:
@chriseth I think we should make this a new PR, so I'll revert this addition, keep the removal from the FAQ document and I'll stash this text for a forthcoming new PR.